yt_dlp/extractor/nova.py

```diff
@@ -45,83 +45,95 @@ def _real_extract(self, url):
+                    video_id, transform_source=js_to_json)
 
-            QUALITIES = ('lq', 'mq', 'hq', 'hd')
-            quality_key = qualities(QUALITIES)
+                QUALITIES = ('lq', 'mq', 'hq', 'hd')
+                quality_key = qualities(QUALITIES)
 
-            for format_id, format_list in bitrates.items():
-                if not isinstance(format_list, list):
-                    format_list = [format_list]
-                for format_url in format_list:
-                    format_url = url_or_none(format_url)
-                    if not format_url:
-                        continue
-                    if format_id == 'hls':
-                        formats.extend(self._extract_m3u8_formats(
-                            format_url, video_id, ext='mp4',
-                            entry_protocol='m3u8_native', m3u8_id='hls',
-                            fatal=False))
-                        continue
-                    f = {
-                        'url': format_url,
-                    }
-                    f_id = format_id
-                    for quality in QUALITIES:
-                        if '%s.mp4' % quality in format_url:
-                            f_id += '-%s' % quality
-                            f.update({
-                                'quality': quality_key(quality),
-                                'format_note': quality.upper(),
-                            })
-                            break
-                    f['format_id'] = f_id
-                    formats.append(f)
+                for format_id, format_list in bitrates.items():
+                    if not isinstance(format_list, list):
+                        format_list = [format_list]
+                    for format_url in format_list:
+                        format_url = url_or_none(format_url)
+                        if not format_url:
+                            continue
+                        if format_id == 'hls':
+                            formats.extend(self._extract_m3u8_formats(
+                                format_url, video_id, ext='mp4',
+                                entry_protocol='m3u8_native', m3u8_id='hls',
+                                fatal=False))
+                            continue
+                        f = {
+                            'url': format_url,
+                        }
+                        f_id = format_id
+                        for quality in QUALITIES:
+                            if '%s.mp4' % quality in format_url:
+                                f_id += '-%s' % quality
+                                f.update({
+                                    'quality': quality_key(quality),
+                                    'format_note': quality.upper(),
+                                })
+                                break
+                        f['format_id'] = f_id
+                        formats.append(f)
```

Comment:
This could probably be refactored with `traverse_obj`, or perhaps `process_format_list` could be generalized more and that could be used here

But since this is old code that you are just indenting, you don't need to change it

```suggestion
                for format_id, format_list in traverse_obj(bitrates, ({dict.items}, ...)):
                    for format_url in traverse_obj(variadic(format_list), lambda _, v: url_or_none(v)):
                        if format_id == 'hls':
                            formats.extend(self._extract_m3u8_formats(
                                format_url, video_id, ext='mp4', m3u8_id='hls', fatal=False))
                            continue
                        f = {'url': format_url}
                        f_id = format_id
                        for quality in QUALITIES:
                            if '%s.mp4' % quality in format_url:
                                f_id += '-%s' % quality
                                f.update({
                                    'quality': quality_key(quality),
                                    'format_note': quality.upper(),
                                })
                                break
                        f['format_id'] = f_id
                        formats.append(f)
```

Reply:
I have just removed the super old code - it's likely it has not been used for over 3 years so there is little point in keeping it imo. However, I kept the last version as a fallback.